portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,21 @@ function addRandomFunFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+function getRandomGreeting() {
+    const greetings =
+      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+
+    // Pick a random greeting.
+    var greeting = greetings[Math.floor(Math.random() * greetings.length)];
+    return greeting;
+}
+
+/**
+ * Fetches data from server and adds it to home page.
+ */
+function onloader(){
+  fetch("/data").then(response => response.json()).then((infoAboutMe) => {
```

Comment:
I think you could simplify this .then chain a bit 
.then(response => response.json()) returns response.json to the next then statement, but since response.json() is just reading a value in the response, you can do it all in one "then" function: 

fetch("/data").then( (response) => {
   const infoAboutMe = response.json();
   document.getElementById(.......) // rest of your code here
});

Let me know if this comment doesn't make sense, kind of hard to talk about then chains over text, happy to chat in person too :)

Reply:
That makes sense thanks for clearing that up!

Reply:
I actually just tested this out and it didn't seem to work. Idk if I'm doing something wrong but wouldn't response.json() return a promise since its asynchronous?

Reply:
My bad, didn't realize response.json() returned a promise as well. In that case, this is exactly right :)